ROADMAP.md

```diff
@@ -87,6 +87,115 @@ We think that it should be done the other way around, i.e. providing a low-level
+
+  ```diff
+  -<TableCell variant="head" classes={{ typeHead: 'typeHead' }} />
+  +<TableCell variant="head" classes={{ head: 'head' }} />
+  ```
+
+- [ ] Remove the `xxxClassName` properties when already covered the `classes` property.
+  These properties were introduced before`classes`.
+  Using a single pattern makes things more predictable and easier to work with.
+
+  ```diff
+  -<Tabs buttonClassName="buttonClassName" indicatorClassName="indicatorClassName" />
+  +<Tabs classes={{ buttonClassName: 'buttonClassName', indicatorClassName: 'indicatorClassName' }} />
+  ```
+
+- [ ] Grid with no spacing by default [#10223](https://github.com/mui-org/material-ui/issues/10223).
+  The negative margin implementation solution currently used comes with [serious limitations](https://material-ui-next.com/layout/grid/#negative-margin).
+  Material-UI is the only library with a non-zero default spacing between the items.
+  Having zero spacing by default will ease the usage of the component.
+
+  ```diff
+  -<Grid />
+  +<Grid spacing={8} />
+  ```
+
+- [ ] Flatten the import path [#9532](https://github.com/mui-org/material-ui/issues/9532).
+  Knowing the component name should be enough for being able to import it.
+
+  ```diff
+  -import CircularProgress from 'material-ui/Progress/CircularProgress';
+  +import CircularProgress from 'material-ui/CircularProgress';
+  ```
+
+  ```diff
+  -import { ListItem } from 'material-ui/List';
+  +import ListItem from 'material-ui/ListItem';
+  ```
+
+- [ ] Change the CSS specificity rule to solve [#10010](https://github.com/mui-org/material-ui/issues/10010) and [#9742](https://github.com/mui-org/material-ui/issues/9742) at scale.
+  It's inspired by the Bootstrap approach to writing CSS. Basically, it follows two rules:
+  1. A variant has one level of specificity. User overrides can be considered a variant, so we can keep things as simple as possible, e.g. `color` or `variant`.
+  2. We increase the specificity for a variant modifier. We already have to do it for the pseudo-classes (`:hover`, `:focus`, etc.). It allows much more control at the cost of extra complexity. Hopefully it makes it more intuitive.
+
+
+  ```diff
+   const styles = {
+  -  checked: {
+  -    color: green[500],
+  +  root: {
+  +    color: green[600],
+  +    '&$checked': {
+  +      color: green[500],
+  +    },
+     },
+  +  checked: {},
+   };
+
+  // ...
+
+     <Checkbox
+       classes={{
+  +      root: classes.root,
+         checked: classes.checked,
+       }}
+     />
+  ```
+
+- [ ] Use `@material-ui npm scope name` #9673. The pros have been raised in the linked issue.
+
+```diff
+-import Button from 'material-ui/Button';
++import Button from '@material-ui/core/Button';
+```
+
+- [ ] Look into the Render Props API over the Component Injection API. This one is an area of investigation. For instance, there is potential for simplifying the customization of the transitions.
```

Comment:
Personally I really love the Component Injection API. Most libraries recently implement a mix of the two by letting the developer use either a `render` prop or a `component` prop. Most also actually implement the Component Injection API while still using `render` as the prop name.

For example see how [`react-tunnels` actually uses the `render` prop](https://github.com/javivelasco/react-tunnels/blob/5d636e3533777aee5d68094db3f1fd29ccd1cf10/src/TunnelPlaceholder.js#L39).

Reply:
https://github.com/final-form/react-final-form#rendering


| Method                          | How it is rendered                                        |
| ------------------------------- | --------------------------------------------------------- |
| `component` prop (CI)                | `return React.createElement(this.props.component, props)` |
| `render` prop                   | `return this.props.render(props)`                         |
| a render function as `children` | `return this.props.children(props)`                       |

Some thoughts:
- I think that we should encourage people using **"a render function as `children`"** over ~"`render` prop"~. It's the same move React did with the new Context API. I'm sure they have made this design choice carefully. I don't know their motivations but mines are:
  - the closing `</` helps with code readability.
  - the `render` property is unnecessary boilerplate.

  It doesn't mean we can't support both.
- The component injection API (CI) has one important limitation ⚠️. When the parent renders, you create a new component, which unmounts all the children. This force you to move the component creation outside of the render method. It's friction. We have documented it in: https://material-ui-next.com/guides/composition/#component-property

Reply:
What I really love is the terseness of `component="nav"` instead of `render={props => <nav {...props} />}` or `…> {props => <nav {...props} />} </…`.

I’d be totally ok if you moved to an hybrid.

What would happen if **both** `component` and a render prop is passed?

Reply:
This point was way more about the `transition` property and `Actions` property of the Table than changing the `component` behavior.